src/H5TSint.c

```diff
@@ -103,35 +112,42 @@ static uint64_t           H5TS_next_thrd_id_s    = 0;
 /* Mutex for access to H5TS_tinfo_next_free_s and H5TS_next_thrd_id_s */
 static H5TS_mutex_t H5TS_tinfo_mtx_s;
 
-/*-------------------------------------------------------------------------
- * Function: H5TS__init
- *
- * Purpose:  Initialize the H5TS interface
- *
- * Return:   Non-negative on success / Negative on failure
- *
- *-------------------------------------------------------------------------
- */
+/*--------------------------------------------------------------------------
+NAME
+   H5TS__init_package -- Initialize interface-specific information
+USAGE
+    herr_t H5TS__init_package()
+RETURNS
+    Non-negative on success/Negative on failure
+DESCRIPTION
+    Initializes any interface-specific data or routines.
+
+--------------------------------------------------------------------------*/
 herr_t
-H5TS__init(void)
+H5TS__init_package(void)
 {
     herr_t ret_value = SUCCEED; /* Return value */
 
-    FUNC_ENTER_PACKAGE_NAMECHECK_ONLY
+    FUNC_ENTER_PACKAGE_NOERR
 
     /* Initialize the global API lock info */
+#ifdef H5_HAVE_THREADSAFE
     if (H5_UNLIKELY(H5TS_mutex_init(&H5TS_api_info_p.api_mutex, H5TS_MUTEX_TYPE_RECURSIVE) < 0))
         HGOTO_DONE(FAIL);
     H5TS_api_info_p.lock_count = 0;
+#else /* H5_HAVE_CONCURRENCY */
```

Comment:
So this code can only be reached if either H5_HAVE_THREADSAFE or H5_HAVE_CONCURRENCY is defined correct? Could we add an assert like directive here? like
#ifndef H5_HAVE_CONCURRENCY
#error
#endif
or just instead of #else:
#elif H5_HAVE_CONCURRENCY
...
#else
#error
#endif

Reply:
Sure, although lots of things won't compile if there's a new concurrency mode introduced.